src/Compute/Compute.Test/ScenarioTests/VirtualMachineTests.ps1

```diff
@@ -2994,6 +2995,66 @@ function Test-VirtualMachineGetStatus
     }
 }
 
+<#
+.SYNOPSIS
+Test Virtual Machines
+#>
+function Test-VirtualMachineGetStatusWithHealhtExtension
+{
+    # Setup
+    $rgname = Get-ComputeTestResourceName
+
+    try
+    {
+        # Common
+        if ($loc -eq $null)
+        {
+            $loc = Get-ComputeVMLocation;
+        }
+        $loc = $loc.Replace(' ', '');
+
+        New-AzResourceGroup -Name $rgname -Location $loc -Force;
+        
+        # VM Profile & Hardware
+        $vmsize = 'Standard_DS2_v2';
+        $vmname = 'vm' + $rgname;
+
+        # OS & Image
+        $username = "admin01";
+        $password = $PLACEHOLDER | ConvertTo-SecureString -AsPlainText -Force;
+        $cred = new-object -typename System.Management.Automation.PSCredential -argumentlist $username, $password;
+        [string]$domainNameLabel = "vcrptestps7691-6f2166";
+        # Virtual Machine
+        New-AzVM -ResourceGroupName $rgname -Location $loc -DomainNameLabel $domainNameLabel -Name $vmname -Credential $cred -Size $vmsize;
+
+        # Adding health extension on VM
+        $publicConfig = @{"protocol" = "http"; "port" = 80; "requestPath" = "/healthEndpoint"};
+        $extensionName = "myHealthExtension"
+        $extensionType = "ApplicationHealthWindows"
+        $publisher = "Microsoft.ManagedServices"
+        Set-AzVMExtension -ResourceGroupName $rgname -VMName $vmname -Publisher $publisher -Settings $publicConfig -ExtensionType $extensionType -ExtensionName $extensionName -Loc $loc -TypeHandlerVersion "1.0"
+
+        # Get VM
+        $vm = Get-AzVM -Name $vmname -ResourceGroupName $rgname -Status;
+
+        # Check for VmHealth Property
+        Assert-NotNull $vm.VMHealth
```

Comment:
Instead of just checking they are not null, can you check for specific values they should be? That would be a more helpful test.

Reply:
The values might change....I think it's due to the deployment of the extension and the status of the vm when the extension is deployed so checking for a specific value would be cumbersome. Anyways, the most important idea for this test is that VMHealth **_is not always returned_** in the properties of the instance view; it is returned only _if there is a health extension_.  VMHealth has some values but the message might be different. This test captures the intended behavior. For your reference, you can read the intended change here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/667. You can also review my demo in the folders in teams. Resolving